dotnet/src/Planners/Planners.Core/Extensions/ReadOnlyFunctionCollectionPlannerExtensions.cs

```diff
@@ -109,20 +147,18 @@ public static class SKContextSequentialPlannerExtensions
     }
 
     private static async Task<IEnumerable<FunctionView>> GetRelevantFunctionsAsync(
-        SKContext context,
         IEnumerable<FunctionView> availableFunctions,
         IAsyncEnumerable<MemoryQueryResult> memories,
+        ILogger? logger = null,
         CancellationToken cancellationToken = default)
     {
-        ILogger? logger = null;
         var relevantFunctions = new ConcurrentBag<FunctionView>();
         await foreach (var memoryEntry in memories.WithCancellation(cancellationToken))
         {
             var function = availableFunctions.FirstOrDefault(x => x.ToFullyQualifiedName() == memoryEntry.Metadata.Id);
             if (function != null)
             {
-                logger ??= context.LoggerFactory.CreateLogger(typeof(SKContext));
-                if (logger.IsEnabled(LogLevel.Debug))
+                if (logger is not null && logger.IsEnabled(LogLevel.Debug))
```

Comment:
Not important right now but it seems like checking whether the logger is enabled should be the logger's job which would reduce checks like this.

Reply:
Yeah agreed -- I think this was related to past feedback from folks like StephenToub though that it's better to do this everywhere to avoid the string allocations when there is no logger and/or it's not enabled for that log level.

Reply:
Obviously we're not doing that everywhere, but yeah, might be worth re-considering in places.